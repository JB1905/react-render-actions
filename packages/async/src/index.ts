export { Async } from './Async';
export { Await } from './Await';
export { Try } from './Try';
export { Catch } from './Catch';
export { Finally } from './Finally';
