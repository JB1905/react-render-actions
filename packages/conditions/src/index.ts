export { If } from './If';
export { Else } from './Else';
export { Then } from './Then';
export { When } from './When';
export { Unless } from './Unless';
