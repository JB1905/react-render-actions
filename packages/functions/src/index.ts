export { Map } from './Map';
export { Filter } from './Filter';
export { Reduce } from './Reduce';
