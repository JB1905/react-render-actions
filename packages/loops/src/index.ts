export { For } from './For';
export { ForOf } from './ForOf';
export { ForIn } from './ForIn';
export { ForEach } from './ForEach';
export { While } from './While';
export { DoWhile } from './DoWhile';
export { Break } from './Break';
export { Continue } from './Continue';
