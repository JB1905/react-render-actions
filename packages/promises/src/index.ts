export { RPromise } from './Promise';
export { Resolve } from './Resolve';
export { Reject } from './Reject';
