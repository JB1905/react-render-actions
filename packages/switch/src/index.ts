export { Switch } from './Switch';
export { Case } from './Case';
export { Default } from './Default';
