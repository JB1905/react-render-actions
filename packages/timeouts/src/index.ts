export { SetTimeout } from './SetTimeout';
export { SetInterval } from './SetInterval';
