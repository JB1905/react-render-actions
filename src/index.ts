export { For } from './Loops/For';
export { ForOf } from './Loops/ForOf';
export { ForIn } from './Loops/ForIn';
export { ForEach } from './Loops/ForEach';
export { While } from './Loops/While';
export { DoWhile } from './Loops/DoWhile';
export { Break } from './Loops/Break';
export { Continue } from './Loops/Continue';

export { SetTimeout } from './Timeouts/SetTimeout';
export { SetInterval } from './Timeouts/SetInterval';

export { Map } from './Functions/Map';
export { Filter } from './Functions/Filter';
export { Reduce } from './Functions/Reduce';
