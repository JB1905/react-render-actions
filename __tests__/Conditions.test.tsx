import React from 'react';
import { render } from '@testing-library/react';

import { If, Else, Then, When, Unless } from '../src';

describe('If-Else', () => {
  it('should render Then', () => {
    const condition = 2 + 2 === 4;

    const { container, getByText } = render(
      <If condition={condition}>
        <Then>two plus two equals four</Then>

        <Else>fail</Else>
      </If>
    );

    expect(container).toMatchSnapshot();

    expect(getByText('two plus two equals four')).toBeDefined();
    expect(getByText('fail')).not.toBeDefined();
  });

  it('should render Else', () => {
    const condition = 2 + 2 === 5;

    const { container, getByText } = render(
      <If condition={condition}>
        <Then>two plus two equals four</Then>

        <Else>fail</Else>
      </If>
    );

    expect(container).toMatchSnapshot();

    expect(getByText('two plus two equals four')).not.toBeDefined();
    expect(getByText('fail')).toBeDefined();
  });
});

describe('When', () => {
  it('should render When', () => {
    const condition = 2 + 2 === 4;

    const { container, getByText } = render(
      <When condition={condition}>
        <p>two plus two equals four</p>
      </When>
    );

    expect(container).toMatchSnapshot();

    expect(getByText('two plus two equals four')).toBeDefined();
  });

  it('should not render When', () => {
    const condition = 2 + 2 === 5;

    const { container, getByText } = render(
      <When condition={condition}>
        <p>two plus two equals four</p>
      </When>
    );

    expect(container).toMatchSnapshot();

    expect(getByText('two plus two equals four')).not.toBeDefined();
  });
});

describe('Unless', () => {
  it('should render Unless', () => {
    const condition = 2 + 2 === 5;

    const { container, getByText } = render(
      <Unless condition={condition}>
        <p>two plus two not equals five</p>
      </Unless>
    );

    expect(container).toMatchSnapshot();

    expect(getByText('two plus two not equals five')).toBeDefined();
  });

  it('should not render Unless', () => {
    const condition = 2 + 2 === 5;

    const { container, getByText } = render(
      <Unless condition={condition}>
        <p>two plus two not equals five</p>
      </Unless>
    );

    expect(container).toMatchSnapshot();

    expect(getByText('two plus two not equals five')).not.toBeDefined();
  });
});
