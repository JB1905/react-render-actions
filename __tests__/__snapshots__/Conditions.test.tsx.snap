// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If-Else should render Else 1`] = `
<div>
  fail
</div>
`;

exports[`If-Else should render Then 1`] = `
<div>
  two plus two equals four
</div>
`;

exports[`Unless should not render Unless 1`] = `
<div>
  <p>
    two plus two not equals five
  </p>
</div>
`;

exports[`Unless should render Unless 1`] = `
<div>
  <p>
    two plus two not equals five
  </p>
</div>
`;

exports[`When should not render When 1`] = `<div />`;

exports[`When should render When 1`] = `
<div>
  <p>
    two plus two equals four
  </p>
</div>
`;
